docs/development/pigs/pig-017.rst

```diff
@@ -0,0 +1,217 @@
+Observatory framework and with the W3C provenance models. In this optic, the development will be done on
+top of `ctapipe.core.provenance`_, re-using its present capabilities when possible and extending it. The
+information captured will relate to the execution environment, the overall high-level interface configuration,
+the workflow execution process (i.e the relationships among datasets and processes and the parameter values
+used in each process step), and the serialisations of the intermediate and final datasets or objects produced.
+
+A simple and non-intrusive implementation strategy should be considered, with minor modifications in the
+code-base of Gammapy. Metaclass inheritance, class decoration and subclassing mechanisms applied to the
+``Analysys`` class of the high-level interface will be explored. The code will be placed in one or a few
+python scripts inside a new folder ``provenance`` at the ``gammapy.utils`` level. The whole solution will
+be mainly based on a provenance description YAML file that will bind tasks of the high-level interface,
+their parameters, datasets used and generated, etc. with the provenance model, its specific semantics and
+the desired provenance capture granularity. In order to procure an accurate provenance description file, a
+good knowledge and understanding of the high-level interface code-base is needed. Needless to say that a
+well established and defined behaviour of the of the high-level interface tasks and configuration settings
+would be desirable before developers actually define the provenance description YAML file.
+
+Serialisation of the captured provenance into log files will be done according to the logging configuration
+settings defined in the high-level interface. A structured format will be defined to record objects used
+and produced, tasks execution, parameters used and relationships among the whole, as time-stamped log
+messages. We will also provide a small set of tools developed with `prov package`_ that will extract the
+provenance information from log files and translate it into standard W3C file formats (i.e. PROV-O RDF,
+PROV-XML and PROV-JSON), as well as to provenance graphs (i.e. PDF, PNG, SVG). The use of a custom
+structured log format allows us to record additional information other than the strictly permitted by
+W3C standards.
+
+All the configuration settings and parameter values may be accessed via the ``Analysis`` class at the very
+moment of the execution of each task. Captured tasks, datasets and hashable objects in general need a unique
+identifier that could be generated with MD5 hashing. The intermediate and final datasets and objects produced
+will be saved into the local disk if this option is enabled as a configuration parameter, and links to the
+serialisations of these objects will be recorded accordingly in the provenance log file. Finally, it is
+advisable to proceed with a general cleaning of the Gammapy code-base so to replace ``print()`` functions
```

Comment:
Remove this statement. We don't have any `print` in Gammapy.
It is the case that we have too few and not well-thought-out `log` statements, so you could make a comment saying that in Gammapy we should improve what it logged, so that users can from log files in most cases understand what analysis was executed and what happened.